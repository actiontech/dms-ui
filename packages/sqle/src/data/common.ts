export {
  ResponseCode,
  CharCode,
  SystemRole,
  ModalSize
} from '@actiontech/shared/lib/enum';

export {
  ModalFormLayout,
  PageBigFormLayout,
  PageFormLayout,
  filterFormButtonLayoutFactory,
  FilterFormColLayout,
  FilterFormLayout,
  FilterFormRowLayout
} from '@actiontech/shared/lib/data/common';

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export enum LoginTypeEnum {
  'ldap' = 'ldap',
  'sqle' = 'sqle'
}

export const SQLE_INSTANCE_SOURCE_NAME = 'SQLE';

export const SqlFiledInitialValue = '/* input your sql */';

export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';

export const WORKFLOW_VERSION_NAME_PATH_KEY = 'versionName';

export const WORKFLOW_VERSION_ID_PATH_KEY = 'versionId';

export const SQL_MANAGEMENT_INSTANCE_PATH_KEY = 'instanceId';

export const SQL_MANAGEMENT_SOURCE_PATH_KEY = 'source';
