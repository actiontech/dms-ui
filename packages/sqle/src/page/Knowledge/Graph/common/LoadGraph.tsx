import {
  useLoadGraph,
  useRegisterEvents,
  useSetSettings,
  useSigma
} from '@react-sigma/core';
import { FC, useEffect } from 'react';
import useGraph from '../hooks/useGraph';
import { EdgeType, NodeType } from '../index.type';
import {
  INodeResponse,
  IEdgeResponse
} from '@actiontech/shared/lib/api/sqle/service/common';

type props = {
  graphData?: {
    nodes: INodeResponse[];
    edges: IEdgeResponse[];
  };
  hoveredNode: string | null;
  setHoveredNode: (node: string | null) => void;
};
const LoadGraph: FC<props> = ({ graphData, hoveredNode, setHoveredNode }) => {
  const { createGraph } = useGraph();
  const sigma = useSigma<NodeType, EdgeType>();
  const registerEvents = useRegisterEvents<NodeType, EdgeType>();
  const setSettings = useSetSettings<NodeType, EdgeType>();
  const loadGraph = useLoadGraph<NodeType, EdgeType>();

  useEffect(() => {
    if (graphData) {
      const graph = createGraph(graphData);
      loadGraph(graph);

      const handleEnterNode = (event: any) => {
        setHoveredNode(event.node);
      };

      const handleLeaveNode = () => {
        setHoveredNode(null);
      };

      registerEvents({
        enterNode: handleEnterNode,
        leaveNode: handleLeaveNode
      });

      return () => {
        // 清理事件监听
        sigma.removeAllListeners('enterNode');
        sigma.removeAllListeners('leaveNode');
      };
    }
  }, [
    graphData,
    createGraph,
    loadGraph,
    registerEvents,
    setHoveredNode,
    sigma
  ]);

  useEffect(() => {
    setSettings({
      nodeReducer: (node, data) => {
        const graph = sigma.getGraph();
        const newData = {
          ...data,
          highlighted: data.highlighted || false
        };

        if (hoveredNode && graph.hasNode(hoveredNode)) {
          try {
            const isNeighbor =
              node === hoveredNode ||
              graph.neighbors(hoveredNode).includes(node);
            if (isNeighbor) {
              newData.highlighted = true;
            } else {
              newData.color = '#E2E2E2';
              newData.highlighted = false;
            }
          } catch (error) {
            // 如果出现错误，保持节点原样
            console.debug('Failed to process node highlighting:', error);
          }
        }
        return newData;
      },
      edgeReducer: (edge, data) => {
        const graph = sigma.getGraph();
        const edgeSize = data.size ?? 3;
        const newData = { ...data, hidden: false, size: edgeSize };

        if (hoveredNode && graph.hasNode(hoveredNode)) {
          try {
            if (graph.extremities(edge).includes(hoveredNode)) {
              newData.size = edgeSize * 1.5;
            } else {
              newData.hidden = true;
            }
          } catch (error) {
            console.debug('Failed to process edge highlighting:', error);
          }
        }
        return newData;
      }
    });
  }, [hoveredNode, setSettings, sigma]);

  return null;
};

export default LoadGraph;
