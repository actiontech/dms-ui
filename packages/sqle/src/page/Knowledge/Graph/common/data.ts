import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
