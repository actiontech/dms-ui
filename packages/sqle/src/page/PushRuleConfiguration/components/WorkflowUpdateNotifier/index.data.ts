import {
  ReportPushConfigListPushUserTypeEnum,
  ReportPushConfigListTriggerTypeEnum
} from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { t } from '../../../../locale';

export const WorkflowPushUserTypeDictionary: Record<
  keyof typeof ReportPushConfigListPushUserTypeEnum,
  string
> = {
  permission_match: t(
    'pushRule.pushRule.workflowUpdateNotifier.permissionMatch'
  ),
  fixed: 'unknown'
};

export const WorkflowPushFrequencyDictionary: Record<
  keyof typeof ReportPushConfigListTriggerTypeEnum,
  string
> = {
  immediately: t('pushRule.pushRule.workflowUpdateNotifier.immediately'),
  timing: 'unknown'
};
