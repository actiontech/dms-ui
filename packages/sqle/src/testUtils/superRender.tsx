import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
