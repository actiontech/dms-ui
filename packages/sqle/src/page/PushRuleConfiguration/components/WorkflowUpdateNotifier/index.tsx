import { Popconfirm, message } from 'antd';
import { useTranslation } from 'react-i18next';
import {
  WorkflowUpdateNotifierFields,
  WorkflowUpdateNotifierProps
} from './index.type';
import {
  WorkflowPushFrequencyDictionary,
  WorkflowPushUserTypeDictionary
} from './index.data';
import { formatTime } from '@actiontech/dms-kit';
import { BasicSwitch, BasicToolTip } from '@actiontech/dms-kit';
import {
  ReadOnlyConfigColumnsType,
  useConfigRender
} from '@actiontech/dms-kit';
import ReportPushConfig from '@actiontech/shared/lib/api/sqle/service/ReportPushConfig';
import { useCurrentProject } from '@actiontech/shared/lib/features';
import { ResponseCode } from '@actiontech/dms-kit';
import { useBoolean, useRequest } from 'ahooks';
import {
  UpdateReportPushConfigReqV1PushUserTypeEnum,
  UpdateReportPushConfigReqV1TriggerTypeEnum
} from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { IReportPushConfigList } from '@actiontech/shared/lib/api/sqle/service/common';
import { useMemo } from 'react';
import { InfoCircleOutlined } from '@actiontech/icons';
import useThemeStyleData from '../../../../hooks/useThemeStyleData';
import {
  PermissionControl,
  PERMISSIONS
} from '@actiontech/shared/lib/features';
const switchFieldName: keyof WorkflowUpdateNotifierFields = 'enabled';
const WorkflowUpdateNotifier: React.FC<WorkflowUpdateNotifierProps> = ({
  config,
  refetch
}) => {
  const { t } = useTranslation();
  const { projectName } = useCurrentProject();
  const [messageApi, messageContextHolder] = message.useMessage();
  const { sharedTheme } = useThemeStyleData();
  const { run: onSubmit, loading: submitPending } = useRequest(
    () =>
      ReportPushConfig.UpdateReportPushConfig({
        project_name: projectName,
        report_push_config_id: config?.report_push_config_id ?? '',
        enabled: !config?.enabled,
        push_user_list: config?.push_user_list,
        push_user_Type:
          config?.push_user_Type as unknown as UpdateReportPushConfigReqV1PushUserTypeEnum,
        trigger_type:
          config?.trigger_type as unknown as UpdateReportPushConfigReqV1TriggerTypeEnum
      }).then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          refetch();
          messageApi.success(
            t('pushRule.pushRule.workflowUpdateNotifier.successTips')
          );
        }
      }),
    {
      manual: true
    }
  );
  const { renderConfigForm } = useConfigRender<WorkflowUpdateNotifierFields>({
    switchFieldName,
    switchFieldLabel: (
      <BasicToolTip
        title={t('pushRule.pushRule.workflowUpdateNotifier.labelTips')}
        suffixIcon={
          <InfoCircleOutlined color={sharedTheme.uiToken.colorTextTertiary} />
        }
      >
        <span>{t('pushRule.pushRule.workflowUpdateNotifier.label')}</span>
      </BasicToolTip>
    )
  });
  const [
    configSwitchPopoverVisible,
    { setTrue: showConfigSwitchPopover, setFalse: closeConfigSwitchPopover }
  ] = useBoolean(false);
  const onConfigSwitchPopoverConfirm = () => {
    closeConfigSwitchPopover();
    onSubmit();
  };
  const readonlyColumnsConfig: ReadOnlyConfigColumnsType<IReportPushConfigList> =
    useMemo(() => {
      return config?.enabled
        ? [
            {
              label: t(
                'pushRule.pushRule.workflowUpdateNotifier.pushFrequency'
              ),
              dataIndex: 'trigger_type',
              render: (type) => {
                return type
                  ? WorkflowPushFrequencyDictionary[type] ?? 'unknown'
                  : '--';
              }
            },
            {
              label: t('pushRule.pushRule.workflowUpdateNotifier.pusher'),
              dataIndex: 'push_user_Type',
              render: (type) => {
                return type
                  ? WorkflowPushUserTypeDictionary[type] ?? 'unknown'
                  : '--';
              }
            },
            {
              label: t('pushRule.pushRule.workflowUpdateNotifier.lastPushTime'),
              dataIndex: 'last_push_time',
              render: (time) => {
                return formatTime(time, '--');
              }
            }
          ]
        : [];
    }, [config?.enabled, t]);
  return (
    <>
      {messageContextHolder}
      {renderConfigForm({
        data: config ?? {},
        columns: readonlyColumnsConfig,
        configExtraButtons: null,
        configField: null,
        configSwitchNode: (
          <PermissionControl
            permission={
              PERMISSIONS.ACTIONS.SQLE.PUSH_RULE_CONFIGURATION
                .WORKFLOW_MODIFICATION_NOTIFIER_SWITCHER_SWITCH
            }
          >
            <Popconfirm
              title={
                config?.enabled
                  ? t(
                      'pushRule.pushRule.workflowUpdateNotifier.closedConfirmTitle'
                    )
                  : t(
                      'pushRule.pushRule.workflowUpdateNotifier.enabledConfirmTitle'
                    )
              }
              open={configSwitchPopoverVisible}
              onConfirm={onConfigSwitchPopoverConfirm}
              onCancel={closeConfigSwitchPopover}
              okText={t('common.ok')}
            >
              <BasicSwitch
                data-testid="config-switch"
                disabled={submitPending}
                onClick={showConfigSwitchPopover}
                checked={!!config?.enabled}
              />
            </Popconfirm>
          </PermissionControl>
        )
      })}
    </>
  );
};
export default WorkflowUpdateNotifier;
