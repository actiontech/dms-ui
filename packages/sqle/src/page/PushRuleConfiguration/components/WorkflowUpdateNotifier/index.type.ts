import { IReportPushConfigList } from '@actiontech/shared/lib/api/sqle/service/common';
import {
  ReportPushConfigListPushUserTypeEnum,
  ReportPushConfigListTriggerTypeEnum
} from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type WorkflowUpdateNotifierProps = {
  config?: IReportPushConfigList;
  permission: boolean;
  refetch: () => void;
};

export type WorkflowUpdateNotifierFields = {
  enabled: boolean;
  triggerType: ReportPushConfigListTriggerTypeEnum;
  pushUserType: ReportPushConfigListPushUserTypeEnum;
};
