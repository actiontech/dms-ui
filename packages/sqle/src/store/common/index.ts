import { ModalStatus } from '@actiontech/dms-kit/es/types/common.type';
import { PayloadAction } from '@reduxjs/toolkit';

export const commonModalReducer = <
  T extends { modalStatus: ModalStatus }
>() => ({
  initModalStatus(
    state: T,
    { payload: { modalStatus } }: PayloadAction<{ modalStatus: ModalStatus }>
  ) {
    state.modalStatus = modalStatus;
  },
  updateModalStatus(
    state: T,
    {
      payload: { modalName, status }
    }: PayloadAction<{ modalName: string; status: boolean }>
  ) {
    if (Object.prototype.hasOwnProperty.call(state.modalStatus, modalName)) {
      state.modalStatus[modalName] = status;
    }
  }
});
