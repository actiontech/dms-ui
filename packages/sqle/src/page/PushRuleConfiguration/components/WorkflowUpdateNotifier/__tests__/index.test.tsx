import { mockUseCurrentProject } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentProject';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import MockReportPushConfigService from '../../../../../testUtils/mockApi/reportPushConfigService';
import { IReportPushConfigList } from '@actiontech/shared/lib/api/sqle/service/common';
import {
  ReportPushConfigListPushUserTypeEnum,
  ReportPushConfigListTriggerTypeEnum
} from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import WorkflowUpdateNotifier from '..';
import { fireEvent, screen } from '@testing-library/dom';
import { act, cleanup } from '@testing-library/react';
import {
  mockProjectInfo,
  mockCurrentUserReturn
} from '@actiontech/shared/lib/testUtil/mockHook/data';
import { mockUsePermission } from '@actiontech/shared/lib/testUtil/mockHook/mockUsePermission';
import { SystemRole } from '@actiontech/shared/lib/enum';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn()
  };
});

describe('test WorkflowUpdateNotifier', () => {
  let updateReportPushConfigSpy: jest.SpyInstance;
  const refetchSpy = jest.fn();
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseCurrentProject();
    mockUseCurrentUser();
    mockUsePermission(undefined, { mockSelector: true });
    updateReportPushConfigSpy =
      MockReportPushConfigService.UpdateReportPushConfig();
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });
  const customRender = (
    config: IReportPushConfigList = {
      report_push_config_id: '1',
      type: 'workflow',
      enabled: true,
      trigger_type: ReportPushConfigListTriggerTypeEnum.immediately,
      push_frequency_cron: '',
      push_user_Type: ReportPushConfigListPushUserTypeEnum.permission_match,
      push_user_list: [],
      last_push_time: '0001-01-01T00:00:00Z'
    }
  ) => {
    return superRender(
      <WorkflowUpdateNotifier config={config} refetch={refetchSpy} />
    );
  };
  it('should render the component with the initial configuration correctly', () => {
    const { container } = customRender();

    expect(container).toMatchSnapshot();
  });

  it('should call the onSubmit function to submit changes when the confirmation popup is confirmed', async () => {
    customRender();

    fireEvent.click(screen.getByTestId('config-switch'));
    await act(async () => jest.advanceTimersByTime(0));

    expect(
      screen.getByText('当前操作将关闭配置，是否继续？')
    ).toBeInTheDocument();

    fireEvent.click(screen.getByText('确 认'));

    expect(updateReportPushConfigSpy).toHaveBeenCalledTimes(1);
    expect(updateReportPushConfigSpy).toHaveBeenCalledWith({
      enabled: false,
      project_name: mockProjectInfo.projectName,
      push_user_Type: 'permission_match',
      push_user_list: [],
      report_push_config_id: '1',
      trigger_type: 'immediately'
    });
    expect(screen.getByTestId('config-switch')).toBeDisabled();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(refetchSpy).toHaveBeenCalledTimes(1);
    expect(screen.getByText('更新成功！')).toBeInTheDocument();

    cleanup();
    updateReportPushConfigSpy.mockClear();

    customRender({
      report_push_config_id: '1',
      type: 'workflow',
      enabled: false,
      trigger_type: ReportPushConfigListTriggerTypeEnum.immediately,
      push_frequency_cron: '',
      push_user_Type: ReportPushConfigListPushUserTypeEnum.permission_match,
      push_user_list: [],
      last_push_time: '0001-01-01T00:00:00Z'
    });

    fireEvent.click(screen.getByTestId('config-switch'));
    await act(async () => jest.advanceTimersByTime(0));

    expect(
      screen.getByText('当前操作将开启配置，是否继续？')
    ).toBeInTheDocument();

    fireEvent.click(screen.getByText('确 认'));

    expect(updateReportPushConfigSpy).toHaveBeenCalledTimes(1);
    expect(updateReportPushConfigSpy).toHaveBeenCalledWith({
      enabled: true,
      project_name: mockProjectInfo.projectName,
      push_user_Type: 'permission_match',
      push_user_list: [],
      report_push_config_id: '1',
      trigger_type: 'immediately'
    });
  });

  it('should disable the switch and hide the confirmation popup when the user lacks permission', () => {
    mockUseCurrentUser({
      ...mockCurrentUserReturn,
      userRoles: {
        ...mockCurrentUserReturn.userRoles,
        [SystemRole.admin]: false,
        [SystemRole.globalManager]: false
      }
    });
    const { container } = customRender({
      report_push_config_id: '1',
      type: 'workflow',
      enabled: false,
      trigger_type: ReportPushConfigListTriggerTypeEnum.immediately,
      push_frequency_cron: '',
      push_user_Type: ReportPushConfigListPushUserTypeEnum.permission_match,
      push_user_list: [],
      last_push_time: '0001-01-01T00:00:00Z'
    });

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId('config-switch')).not.toBeInTheDocument();
  });
});
