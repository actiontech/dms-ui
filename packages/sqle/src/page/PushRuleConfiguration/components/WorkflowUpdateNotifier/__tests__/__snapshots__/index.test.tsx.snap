// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test WorkflowUpdateNotifier should disable the switch and hide the confirmation popup when the user lacks permission 1`] = `
<div>
  <section
    class="css-zupqof"
  >
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="switch-field-wrapper"
      >
        <div
          class="ant-form-item has-label-tip css-1jlm9cy css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left css-dev-only-do-not-override-txh9fw"
            >
              <label
                class="ant-form-item-no-colon"
                title=""
              >
                <div
                  class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <span>
                      工单变更推送
                    </span>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <svg
                      color="#8a8f99"
                      height="16"
                      viewBox="0 0 14 14"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 12.833A5.833 5.833 0 1 1 7 1.167a5.833 5.833 0 0 1 0 11.666m0-1.166a4.666 4.666 0 1 0 0-9.333 4.666 4.666 0 0 0 0 9.333m-.583-7.584h1.166V5.25H6.417zm0 2.334h1.166v3.5H6.417z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </label>
            </div>
            <div
              class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="config-field-wrapper"
        hidden=""
      />
    </form>
  </section>
</div>
`;

exports[`test WorkflowUpdateNotifier should render the component with the initial configuration correctly 1`] = `
<div>
  <section
    class="css-zupqof"
  >
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="switch-field-wrapper"
      >
        <div
          class="ant-form-item has-label-tip css-1jlm9cy css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left css-dev-only-do-not-override-txh9fw"
            >
              <label
                class="ant-form-item-no-colon"
                title=""
              >
                <div
                  class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <span>
                      工单变更推送
                    </span>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <svg
                      color="#8a8f99"
                      height="16"
                      viewBox="0 0 14 14"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 12.833A5.833 5.833 0 1 1 7 1.167a5.833 5.833 0 0 1 0 11.666m0-1.166a4.666 4.666 0 1 0 0-9.333 4.666 4.666 0 0 0 0 9.333m-.583-7.584h1.166V5.25H6.417zm0 2.334h1.166v3.5H6.417z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </label>
            </div>
            <div
              class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                    >
                      <button
                        aria-checked="true"
                        class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw ant-switch-checked"
                        data-testid="config-switch"
                        role="switch"
                        type="button"
                      >
                        <div
                          class="ant-switch-handle"
                        />
                        <span
                          class="ant-switch-inner"
                        >
                          <span
                            class="ant-switch-inner-checked"
                          />
                          <span
                            class="ant-switch-inner-unchecked"
                          />
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="config-field-wrapper"
        hidden=""
      />
    </form>
    <div
      class="ant-descriptions system-config-view-wrapper css-cs8tvb css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-descriptions-view"
      >
        <table>
          <tbody>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="3"
              >
                <div
                  class="ant-descriptions-item-container"
                >
                  <span
                    class="ant-descriptions-item-label"
                  >
                    推送频率
                  </span>
                  <span
                    class="ant-descriptions-item-content"
                  >
                    根据工单状态变更即时触发
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="3"
              >
                <div
                  class="ant-descriptions-item-container"
                >
                  <span
                    class="ant-descriptions-item-label"
                  >
                    推送人
                  </span>
                  <span
                    class="ant-descriptions-item-content"
                  >
                    根据权限匹配有操作权限的成员
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="3"
              >
                <div
                  class="ant-descriptions-item-container"
                >
                  <span
                    class="ant-descriptions-item-label"
                  >
                    最近一次推送时间
                  </span>
                  <span
                    class="ant-descriptions-item-content"
                  >
                    1-01-01 08:05:43
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </section>
</div>
`;
